Extensions/blacklist.js

```diff
@@ -551,7 +551,9 @@ XKit.extensions.blacklist = new Object({
 
 				m_content = XKit.tools.replace_all(m_content, "&nbsp;", " ");
 				m_content = m_content.toLowerCase();
-
+				
+			    // Preserve href links.
+			    m_content = m_content.replace(/<a\s+(?:[^>]*?\s+)?href="([^"]*)".*?>/gm, "$1" + ' ');
```

Comment:
The `"$1" + ' '` should probably be written as `' $1 '` to make sure that it doesn't get concatenated with whatever is in front of it.

For example, `apple<a href="banana">canteloupe</a>` would transform into "applebananacanteloupe" which would fail to be blacklisted by "banana" or "canteloupe" (if the Blacklist function behaves as documented, which I doubt)